### DISCUSSION TITLE
[Request] Release UTM SE on TestFlight or App Store

### DISCUSSION THREAD

**Discussion by kkebo** [utmapp/UTM#2561] [Ideas]
@osy said

> TestFlight only allows distribution certificates and not development certificates. Distribution certificates do not have get-task-allow entitlement.

in #42, but [UTM SE doesn't seem to need the `get-task-allow` entitlement](https://github.com/utmapp/UTM/blob/306e663871b8a9bae8b9ea47329fb6e51c4a8956/scripts/package.sh#L225).

If so, how about releasing UTM SE on TestFlight or App Store? I know it's slower than UTM, but it's worth it.

**Comment by osy** [comment]
Apple doesn’t allow it because it’s an emulator.

**Reply by kkebo** [reply]
@osy Thank you for replying.

I know that game console emulators are banned, but is there any precedent for PC emulators? I'm wondering which clause of [the guidelines](https://developer.apple.com/app-store/review/guidelines/) it falls under.

**Reply by osy** [reply]
Don’t ask me ask apple

**Reply by kkebo** [reply]
Okay, so have you ever had UTM reviewed by Apple? Is there any evidence that Apple doesn't allow it?

**Reply by osy** [reply]
You think I'm just speculating ;)

**Reply by kkebo** [reply]
To be honest, you're right. However, I'm sorry if I made you uncomfortable, I didn't mean to.

**Reply by sinoru** [reply]
I don't know well either, but iSH has been passed the review. It might be applied to UTM SE?

**Reply by osy** [reply]
@sinoru Apple rejections don’t make sense to me either but there’s no point in telling me this…

**Reply by sinoru** [reply]
@osy So, even TestFlight External Review has been rejected?

**Reply by ghost** [reply]
> Apple doesn’t allow it because it’s an emulator.
@osy 
someone has uploaded "UTM SE" to testflight
https://testflight.apple.com/join/yAy7ZyZx

**Reply by kkebo** [reply]
If Apple approves UTM SE for the TestFlight, why doesn't @osy submit it?

**Reply by osy** [reply]
Apple blocked it.

**Reply by kkebo** [reply]
Thanks for replying. Does that mean it will be approved or disapproved from time to time?

**Reply by osy** [reply]
Apple. Blocked. It.

**Reply by kkebo** [reply]
It's actually approved [here](https://github.com/utmapp/UTM/discussions/2561#discussioncomment-1242419), so I'm curious to know the difference from when you got rejected.

**Reply by osy** [reply]
![PNG image](https://user-images.githubusercontent.com/50960678/134777587-e2ff4fbf-8de0-4b0e-994e-d7c2d7b221c5.png)

**Reply by kkebo** [reply]
Thank you for sharing the details. Is the reason that it's an emulator just like before?

**Reply by osy** [reply]
Sigh
